main.go

```diff
@@ -22,51 +22,74 @@ func main() {
 
 	// Iterate over each pull request
 	for _, pr := range prs {
-		fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
-
-		fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
-		diff, err := client.GetPrDiff(pr)
-		if err != nil {
-			log.Fatalf("Failed to get raw diff: %v", err)
+		comments := reviewPR(pr, client)
+		commitId := pr.GetHead().GetSHA()
+		for _, comment := range comments {
+			// post to github
+			fmt.Printf("Posting comment to PR #%d\n", pr.GetNumber())
+			ghComment := client.ParsedCommentToGithubComment(comment, commitId)
+			err := client.PostComment(pr, ghComment)
+			if err != nil {
+				fmt.Printf("Failed to post comment: %v\n", err)
+				fmt.Printf("Comment: %v\n", ghComment)
+			}
 		}
+	}
+}
+
+func reviewPR(pr *github.PullRequest, client github_client.GithubRepoClient) []*comments.Comment {
+	fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
+
+	fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
+	diff, err := client.GetPrDiff(pr)
+	if err != nil {
+		log.Fatalf("Failed to get raw diff: %v", err)
+	}
+
+	diffTransformer := transformer.DiffTransformer{}
+	diffTransformer.Transform(diff)
 
-		diffTransformer := transformer.DiffTransformer{}
-		diffTransformer.Transform(diff)
+	fileSegments := diffTransformer.GetFileSegments()
+	allComments := []*comments.Comment{}
+	// failedComments := []string{}
 
-		segments := diffTransformer.GetSegments()
-		allComments := []string{}
-		fmt.Printf("Getting commments for %d segments\n", len(segments))
+	fmt.Printf("Getting comments for %d segments\n", len(fileSegments))
+	for filename, segments := range fileSegments {
 		for _, segment := range segments {
 			comment := retrieveComments(segment)
-			allComments = append(allComments, comment)
+			if comment == nil {
+				// failedComments = append(failedComments, segment)
+				continue
+			}
+			zippedComments, err := comments.ZipComment(segment, *comment, filename)
+			if err != nil {
+				log.Fatalf("Failed to zip comment: %v", err)
+			}
+			allComments = append(allComments, zippedComments...)
+			if len(allComments) >= 1 {
+				break
+			}
```

Comment:
Suggestion: consider breaking out of both loops (fileSegments loop and the segment loop) when the comment is posted successfully to save time.